SegmentEditorSplitVolume/SegmentEditorSplitVolumeLib/SegmentEditorEffect.py

```diff
@@ -147,10 +168,14 @@ def onApply(self):
     inputVolumeParentItem = shNode.GetItemParent(shNode.GetItemByDataNode(inputVolume))
     outputShFolder = shNode.CreateFolderItem(inputVolumeParentItem, inputVolume.GetName()+" split")
 
-    # Iterate over segments
+    # Iterate over visible segments
     slicer.app.setOverrideCursor(qt.Qt.WaitCursor)
-    for segmentIndex in range(segmentationNode.GetSegmentation().GetNumberOfSegments()):
-      segmentID = segmentationNode.GetSegmentation().GetNthSegmentID(segmentIndex)
+    inputSegmentIDs = vtk.vtkStringArray()
+    segmentationNode.GetDisplayNode().GetVisibleSegmentIDs(inputSegmentIDs)
+    for segmentIndex in range(inputSegmentIDs.GetNumberOfValues()):

```

Comment:
I've just realized that this does not work correctly if the current segment is not set to visible. Something like this should work:

```
if applyToAllVisibleSegments:
    inputSegments = []
    for segmentIndex in range(inputSegmentIDs.GetNumberOfValues()):
        inputSegments.append(inputSegmentIDs.GetValue(segmentIndex))
else:
    inputSegments = [currentSegmentID]

for segmentID in inputSegments:
    segmentIDs = vtk.vtkStringArray()
    ...
```

I know that I recommended some other effect and you used that as an example, so that effect would probably need to be updated, too.

Reply:
Split volume works now on all visible segments, or only selected segment on request, irrespective of its visibility.